scheduler_test.go

```diff
@@ -899,6 +899,26 @@ func TestScheduler_NewJobTask(t *testing.T) {
 			NewTask(&testFuncWithParams, "one", "two"),
 			nil,
 		},
+		{
+			"parameter type does not match - different argument types against variadic parameters",
+			NewTask(func(args ...string) {}, "one", 2),
+			ErrNewJobWrongTypeOfParameters,
+		},
+		{
+			"all good string - variadic",
+			NewTask(func(args ...string) {}, "one", "two"),
+			nil,
+		},
+		{
+			"all good struct - variadic",
+			NewTask(func(args ...interface{}) {}, struct{}{}),
+			nil,
+		},
+		{
+			"all good no arguments passed in - variadic",
+			NewTask(func(args ...interface{}) {}),
```

Comment:
could you add a test case with multiple argument types where the last one is variadic? 

something like
```golang
NewTask(func(arg int, args ...string) {}, 1, "one", "two")
```

Reply:
Sure this was neglected in my previous commit. And it's added now in this PR. Please have a look.